evm/src/cpu/kernel/asm/memory/metadata.asm

```diff
@@ -312,3 +312,15 @@ global sys_basefee:
     %mload_context_metadata(@CTX_METADATA_STATIC)
     %jumpi(fault_exception)
 %endmacro
+
+%macro add_or_fault
```

Comment:
Hm I was imagining an implementation like

```
sum = x + y
if sum < x
    fault // overflow
return sum
```

I think this implementation is fine too, just less general in a way since `%ensure_reasonable_offset` is about memory offsets/lengths. Maybe we don't need the extra generality, but do you see any downside to the overflow-check approach?

Reply:
I also prefer the overflow-check approach since in some cases `offset` can be arbitrarily large if `size` is zero.